apisix/core/etcd.lua

```diff
@@ -44,24 +73,143 @@ end
 _M.new = new
 
 
+local function kvs2node(kvs)
+    local node = {}
+    node.key = kvs.key
+    node.value = kvs.value
+    node.createdIndex = tonumber(kvs.create_revision)
+    node.modifiedIndex = tonumber(kvs.mod_revision)
+    return node
+end
+
+
+local function notfound(res)
+    res.body.message = "Key not found"
+    res.reason = "Not found"
+    res.status = 404
+    return res
+end
+
+
+function _M.postget(res, realkey)
+    if res.body.error == "etcdserver: user name is empty" then
+        return nil, "insufficient credentials code: 401"
+    end
+
+    res.headers["X-Etcd-Index"] = res.body.header.revision
+
```

Comment:
To be honest, I don’t understand these codes

Reply:
It is to transfer the etcd v3 output structure to the same as the original v2, to keep all the interfaces the same.
Maybe they need to have a better function name

Reply:
I changed:
`postget`      ->  `get_res_format`
`postwatch`  ->  `watch_res_format`

Maybe that would be more readable